test/prototype_transforms_dispatcher_infos.py

```diff
@@ -1,39 +1,89 @@
+import collections.abc
 import dataclasses
-from typing import Callable, Dict, Sequence, Type
+from typing import Callable, Dict, Optional, Sequence, Type
 
 import pytest
 import torchvision.prototype.transforms.functional as F
-from prototype_transforms_kernel_infos import KERNEL_INFOS, Skip
+from prototype_common_utils import BoundingBoxLoader
+from prototype_transforms_kernel_infos import KERNEL_INFOS, KernelInfo, Skip
 from torchvision.prototype import features
 
 __all__ = ["DispatcherInfo", "DISPATCHER_INFOS"]
 
-KERNEL_SAMPLE_INPUTS_FN_MAP = {info.kernel: info.sample_inputs_fn for info in KERNEL_INFOS}
+KERNEL_INFO_MAP = {info.kernel: info for info in KERNEL_INFOS}
+
+
+@dataclasses.dataclass
+class PILKernelInfo:
+    kernel: Callable
+    kernel_name: str = dataclasses.field(default=None)
+
+    def __post_init__(self):
+        self.kernel_name = self.kernel_name or self.kernel.__name__
 
 
 @dataclasses.dataclass
 class DispatcherInfo:
     dispatcher: Callable
     kernels: Dict[Type, Callable]
+    kernel_infos: Dict[Type, KernelInfo] = dataclasses.field(default=None)
+    pil_kernel_info: Optional[PILKernelInfo] = None
+    method_name: str = dataclasses.field(default=None)
     skips: Sequence[Skip] = dataclasses.field(default_factory=list)
     _skips_map: Dict[str, Skip] = dataclasses.field(default=None, init=False)
 
     def __post_init__(self):
+        self.kernel_infos = {feature_type: KERNEL_INFO_MAP[kernel] for feature_type, kernel in self.kernels.items()}
+        self.method_name = self.method_name or self.dispatcher.__name__
         self._skips_map = {skip.test_name: skip for skip in self.skips}
 
-    def sample_inputs(self, *types):
-        for type in types or self.kernels.keys():
-            if type not in self.kernels:
-                raise pytest.UsageError(f"There is no kernel registered for type {type.__name__}")
+    def sample_inputs(self, *feature_types, filter_metadata=True):
+        for feature_type in feature_types or self.kernels.keys():
+            if feature_type not in self.kernels:
+                raise pytest.UsageError(f"There is no kernel registered for type {feature_type.__name__}")
+
+            sample_inputs = self.kernel_infos[feature_type].sample_inputs_fn()
+            if not filter_metadata:
+                yield from sample_inputs
+            else:
+                for args_kwargs in sample_inputs:
+                    for attribute in feature_type.__annotations__.keys():
+                        if attribute in args_kwargs.kwargs:
+                            del args_kwargs.kwargs[attribute]
```

Comment:
This removes metadata like `image_size` or `format` from the sample inputs. They are only there in the first place since we are using the sample inputs of the kernels for the dispatcher.